src/pytest_molecule/__init__.py

```diff
@@ -124,17 +125,17 @@ def pytest_configure(config):
                 # selinux bindings are not guaranteed to fail molecule execution.
 
 
-def pytest_collect_file(parent, path) -> Optional["Node"]:
+def pytest_collect_file(file_path: Path, parent) -> Optional["Node"]:
     """Transform each found molecule.yml into a pytest test."""
 
     # We do not want to recognize paths with symlinks as valid
-    if os.path.realpath(path) != path:
+    if file_path.is_symlink():
         return None
 
-    if path.basename == "molecule.yml":
+    if file_path.name == "molecule.yml":
         if hasattr(MoleculeFile, "from_parent"):
-            return MoleculeFile.from_parent(fspath=path, parent=parent)
-        return MoleculeFile(path, parent)
+            return MoleculeFile.from_parent(path=file_path, parent=parent)
+        return MoleculeFile(file_path, parent)
```

Comment:
@ssbarnea i have two possibilities to fix that mypy finding as per my understanding:

```
src/pytest_molecule/__init__.py:138: error: Argument 1 to "MoleculeFile" has incompatible type "Path"; expected "Optional[local]"
```

1) Explicitly only allow pytest greater 5.4 to have "from_parent" available which allows us to simplify these three lines 136-138 to:

```
return MoleculeFile.from_parent(path=file_path, parent=parent)
```

5.4 was released March 2020.

2) We retain compatibility with pytest < 5.4 through an conditional import of py.path.local to convert the file_path from Pathlib.Path to py.path.local.

Which route do you want to go?

Thanks in advance!

Best Regards, Uli

Reply:
Bump minimum pytest version requires to 5.4. That release is more than two years old. If someone wants to use older ones they should for it ;)

Reply:
Actually it gets a little worse. Due to the changed method definition (especially the typing) of pytest_collect_file pytest<7 doesn't like it anymore (I only tested against pytest 7 so far so didn't come across that). I'll check tomorrow how the compatibility with pytest 6 could be retained - I have an idea.